Edits to contributing.md

jhaiduce opened the pull request:
This PR introduces the following minor edits to contributing.md:

- Added a paragraph informing people that all the python versions in tox.ini must be installed and in the user's $PATH in order to run the full test suite
- Added instructions for installing gettext on Fedora
- Changed "Linux (Debian)" to simply "Linux" in the comment preceding the instructions for installing geckodriver, since the instructions provided should work on any Linux that has wget and tar and runs on the x86_64 architecture.

stevepiercy commented:
@jhaiduce I overhauled the "Using tox" section because the original needed some lovin'. Please take a look at my commit, and let me know if this makes things more explicit and clear. Thank you!

jhaiduce commented:
Made some minor edits to the new "Setting up a development environment" section:

- Moved quote marks outward so they contain the brackets (in case someone is using a shell that treats brackets are syntactical)
- Added a couple words to the opening sentence to explicitly mention the venv module (so if a reader is unfamiliar with venv they have a better idea what to google).

stevepiercy commented on contributing.md:
```diff
@@ -22,8 +22,8 @@ All features must be documented with code samples in narrative documentation, AP
 
-Create a virtual environment.
-Into this virtual environment, install Deform development requirements for unit testing, lint, docs, and functional testing with a single command.
+Create a virtual environment using Python's venv module.
```
Good idea. I'll enhance this a bit with a new commit.

stevepiercy commented:
Assuming the build passes, I'll merge it. Thank you!
